File: src/main/java/io/github/chaosdave34/kitpvp/commands/GommeModeCommand.java
package io.github.chaosdave34.kitpvp.commands;

import io.github.chaosdave34.kitpvp.ExtendedPlayer;
import org.bukkit.GameMode;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class GommeModeCommand implements CommandExecutor {
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (sender instanceof Player p) {
            ExtendedPlayer extendedPlayer = ExtendedPlayer.from(p);
            extendedPlayer.setGameState(ExtendedPlayer.GameState.DEBUG);
            p.setGameMode(GameMode.CREATIVE);
            return true;
        }

        return false;
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/commands/GommemodeCommand.kt
package io.github.chaosdave34.kitpvp.commands

import com.mojang.brigadier.Command
import io.github.chaosdave34.kitpvp.ExtendedPlayer
import io.papermc.paper.command.brigadier.CommandSourceStack
import io.papermc.paper.command.brigadier.Commands
import org.bukkit.GameMode
import org.bukkit.entity.Player

object GommemodeCommand {
    fun register(commands: Commands) {
        commands.register(
            Commands.literal("gommemode")
                .requires { it.sender.isOp }
                .requires { it.sender is Player }
                .executes { enterGommemode(it.source) }
                .build()
        )
    }

    private fun enterGommemode(source: CommandSourceStack): Int {
        val player = source.sender as Player
        val extendedPlayer = ExtendedPlayer.from(player)
        extendedPlayer.gameState = ExtendedPlayer.GameState.DEBUG
        player.gameMode = GameMode.CREATIVE

        return Command.SINGLE_SUCCESS
    }
}